src/main/java/org/terasology/logic/health/HealthCommands.java

```diff
@@ -72,6 +72,51 @@ public String damageWithType(@Sender EntityRef client, @CommandParam("damageType
             return "Specified damage type does not exist.";
         }
     }
+    @Command(shortDescription = "Damage Resistance", runOnServer = true,
+            requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String damageResist(@CommandParam("DamageType") String dType,@CommandParam("Percentage") int value,@Sender EntityRef clientEntity) {
+        if(value>=0&&value<=100) {
+            ClientComponent player = clientEntity.getComponent(ClientComponent.class);
+            DamageResistComponent damageResist = player.character.getComponent(DamageResistComponent.class);
+            String s="ALL";
+            if(damageResist==null) {
+                player.character.addComponent(new DamageResistComponent());
+                damageResist = player.character.getComponent(DamageResistComponent.class);
+            }
+            if(dType.equals(s)){
+                damageResist.setAll(value);
+                if(value==0){
+                    player.character.removeComponent(DamageResistComponent.class);
+                }
+            }
+
+            if(damageResist.damageTypes.containsKey(dType)){
```

Comment:
This is fine for this PR, but once this PR is merged I'd like to see an issue opened about making this more dynamic as to allow for damage types beyond the few listed in the `setAll` method. That way, if a module decided to say add "psychic" as a damage type then it would be automatically handled by the system.

Like I said, not something that needs to be changed now, but I just wanted to make a note about it :)

Reply:
Yeah sure.And maybe If I have time I'll look into it myself. 👍